### DISCUSSION TITLE
Compatibility with CUDA 12

### DISCUSSION THREAD

**Discussion by mahmoodn** [accel-sim/accel-sim-framework#316] [General]
Hi,
Just want to know if there is any general issue with creating traces with CUDA 12.x and simulate them with Accel-sim or not. When I set CUDA_INSTALL_PATH to CUDA 12, I get the error that CUDA version is unsupported (while sourcing environment). So, I am curious to know if that is real incompatibility issue to just a message that current code has not been tested with CUDA 12.

P.S: I am not talking about new instruction set found in new architectures.

**Comment by JRPan** [comment]
We are waiting for NVbit to support CUDA 12. 
Except the tracer, it should work. Even though we did not test it.

**Reply by ymingX** [reply]
Now NVbit has supported CUDA 12. But GPGPU-sim doesn't support yet. It seems because it use `textureReference`, which has been removed from CUDA 12.0.
See [https://github.com/gpgpu-sim/gpgpu-sim_distribution/issues/315](url) and [https://forums.developer.nvidia.com/t/cuda-12-0-still-support-for-texture-reference-support-for-pascal-architecture-warp-synchronous-programming/237284](url)

**Comment by mahmoodn** [comment]
Good to know that NVbit 1.7 has been released...

**Comment by tgrogers** [comment]
This is fixed in the latest dev

**Reply by yang-11** [reply]
Hi, I encounter a problem. My device is RTX 5080 and cuda version is 12.8.
I find when I use  command 
`./util/job_launching/run_simulations.py -B rodinia_2.0-ft -C QV100-PTX -N myTest-PTX`
the tests will be run and finished successfully, but when I use 
`/gpu-simulator/bin/release/accel-sim.out -trace ./hw_run/rodinia_2.0-ft/9.1/backprop-rodinia-2.0-ft/4096___data_result_4096_txt/traces/kernelslist.g -config ./gpu-simulator/gpgpu-sim/configs/tested-cfgs/SM7_QV100/gpgpusim.config -config ./gpu-simulator/configs/tested-cfgs/SM7_QV100/trace.config`
the test will report "unsupported binary version: 120"
Why the run_simulation.py work but the accel-sim.out not?

**Comment by yang-11** [comment]
I encounter a problem. My device is RTX 5080 and cuda version is 12.8.
I find when I use command
./util/job_launching/run_simulations.py -B rodinia_2.0-ft -C QV100-PTX -N myTest-PTX
the tests will be run and finished successfully, but when I use
/gpu-simulator/bin/release/accel-sim.out -trace ./hw_run/rodinia_2.0-ft/9.1/backprop-rodinia-2.0-ft/4096___data_result_4096_txt/traces/kernelslist.g -config ./gpu-simulator/gpgpu-sim/configs/tested-cfgs/SM7_QV100/gpgpusim.config -config ./gpu-simulator/configs/tested-cfgs/SM7_QV100/trace.config
the test will report "unsupported binary version: 120"
Why the run_simulation.py work but the accel-sim.out not?

**Reply by JRPan** [reply]
`QV100-PTX` run in PTX mode, execution-driven. It's like running gpgpu-sim, without the frontend. 
As long as it has ptx for the gencode included during compilation, it works. 

Running `./util/job_launching/run_simulations.py -B rodinia_2.0-ft -C QV100-SASS -N myTest-PTX` should give you the same error, unsupported binary version

**Reply by yang-11** [reply]
Running ./util/job_launching/run_simulations.py -B rodinia_2.0-ft -C QV100-SASS -N myTest will show "All 10 Tests Done"
This phenomenon made me confused

**Reply by yang-11** [reply]
The running flow is below

root@5a2ea1476b11:~/accel-sim-framework-dev# ./util/job_launching/run_simulations.py -B rodinia_2.0-ft -C QV100-SASS -T ./hw_run/traces/device-0/12.8/ -N my_test
Cannot find a supported job management system. Spawning jobs locally.
Running Simulations with GPGPU-Sim built from 
accelsim-commit-_modified_0.0_25-12-10-18-07-32gpgpu-sim_git-commit-_modified_0.0
 
Using configs: QV100-SASS
Benchmark: rodinia_2.0-ft
Run Subdir = QV100-SASS
Parameters = 
#SASS
#SASS-Driven Accel-Sim

Base config file = /root/accel-sim-framework-dev/gpu-simulator/gpgpu-sim/configs/tested-cfgs/SM7_QV100/gpgpusim.config
Job 183 queued (backprop-rodinia-2.0-ft-4096___data_result_4096_txt QV100-SASS)
Job 184 queued (bfs-rodinia-2.0-ft-__data_graph4096_txt___data_graph4096_result_txt QV100-SASS)
Job 185 queued (hotspot-rodinia-2.0-ft-30_6_40___data_result_30_6_40_txt QV100-SASS)
Job 186 queued (heartwall-rodinia-2.0-ft-__data_test_avi_1___data_result_1_txt QV100-SASS)
Job 187 queued (lud-rodinia-2.0-ft-_v__b__i___data_64_dat QV100-SASS)
Job 188 queued (nw-rodinia-2.0-ft-128_10___data_result_128_10_txt QV100-SASS)
Job 189 queued (nn-rodinia-2.0-ft-__data_filelist_4_3_30_90___data_filelist_4_3_30_90_result_txt QV100-SASS)
Job 190 queued (pathfinder-rodinia-2.0-ft-1000_20_5___data_result_1000_20_5_txt QV100-SASS)
Job 191 queued (srad_v2-rodinia-2.0-ft-__data_matrix128x128_txt_0_127_0_127__5_2___data_result_matrix128x128_1_150_1_100__5_2_txt QV100-SASS)
Job 192 queued (streamcluster-rodinia-2.0-ft-3_6_16_1024_1024_100_none_output_txt_1___data_result_3_6_16_1024_1024_100_none_1_txt QV100-SASS)
ProcMan spawned [pid=605842]
root@5a2ea1476b11:~/accel-sim-framework-dev# nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2025 NVIDIA Corporation
Built on Wed_Jan_15_19:20:09_PST_2025
Cuda compilation tools, release 12.8, V12.8.61
Build cuda_12.8.r12.8/compiler.35404655_0
root@5a2ea1476b11:~/accel-sim-framework-dev# ./util/job_launching/monitor_func_test.py -v -N my_test
Calling job_status.py
Using logfiles ['/root/accel-sim-framework-dev/util/job_launching/../job_launching/logfiles/sim_log.my_test.25.12.10-Wednesday.txt']
procman.id      Node                            App                     AppArgs                 Version                 Config          RunningTime     Mem     JobStatus                       Basic GPGPU-Sim Stats
183             UNKNOWN                         backprop-rodinia-2.0    4096___data_result_4    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
184             UNKNOWN                         bfs-rodinia-2.0-ft      __data_graph4096_txt    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
185             UNKNOWN                         hotspot-rodinia-2.0-    30_6_40___data_resul    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
186             UNKNOWN                         heartwall-rodinia-2.    __data_test_avi_1___    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
187             UNKNOWN                         lud-rodinia-2.0-ft      _v__b__i___data_64_d    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
188             UNKNOWN                         nw-rodinia-2.0-ft       128_10___data_result    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
189             UNKNOWN                         nn-rodinia-2.0-ft       __data_filelist_4_3_    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
190             UNKNOWN                         pathfinder-rodinia-2    1000_20_5___data_res    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
191             UNKNOWN                         srad_v2-rodinia-2.0-    __data_matrix128x128    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
192             UNKNOWN                         streamcluster-rodini    3_6_16_1024_1024_100    accelsim-commit-_mod    QV100-SASS      UNKNOWN         UNKNOWN COMPLETE_NO_OTHER_INFO
Passed:0/10, No error:10/10, Failed/Error:0/10, Running:0/10, Waiting:0/10
All 10 Tests Done.
Congratulations! All Tests Pass!
root@5a2ea1476b11:~/accel-sim-framework-dev#

**Reply by yang-11** [reply]
I find the output in sim_run_12.8/ show “unsupported binary version”，it's just not showed in the terminal.
